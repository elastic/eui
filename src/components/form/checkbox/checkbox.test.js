import React from 'react';
import { render } from 'enzyme';
import {
  requiredProps,
  startThrowingReactWarnings,
  stopThrowingReactWarnings,
} from '../../../test';

import {
  EuiCheckbox,
  TYPES,
} from './checkbox';

beforeAll(startThrowingReactWarnings);
afterAll(stopThrowingReactWarnings);

const checkboxRequiredProps = {
  id: 'id',
  onChange: () => {},
};

describe('EuiCheckbox', () => {
  test('is rendered', () => {
    const component = render(
      <EuiCheckbox
        id="id"
        onChange={() => {}}
        {...requiredProps}
      />
    );

    expect(component)
      .toMatchSnapshot();
  });

  describe('props', () => {
    test('id is required', () => {
      expect(
        () => <EuiCheckbox checked={true} onChange={() => {}}/>
      ).toThrow();
    });

    test('onChange is required', () => {
      expect(
        () => <EuiCheckbox id="id" checked={true}/>
      ).toThrow();
    });

    test('check is rendered', () => {
      const component = render(
        <EuiCheckbox
          {...checkboxRequiredProps}
          checked
        />
      );

      expect(component)
        .toMatchSnapshot();
    });

    test('label is rendered', () => {
      const component = render(
        <EuiCheckbox
          {...checkboxRequiredProps}
          label={<span>Label</span>}
        />
      );

      expect(component)
        .toMatchSnapshot();
    });

    describe('type', () => {
      TYPES.forEach(value => {
        test(`${value} is rendered`, () => {
          const component = render(
            <EuiCheckbox
              {...checkboxRequiredProps}
              type={value}
            />
          );

          expect(component)
            .toMatchSnapshot();
        });
      });
    });

    describe('disabled', () => {
      test(`disabled is rendered`, () => {
        const component = render(
          <EuiCheckbox
            {...checkboxRequiredProps}
            disabled
          />
        );

        expect(component)
          .toMatchSnapshot();
      });
    });
  });
});
