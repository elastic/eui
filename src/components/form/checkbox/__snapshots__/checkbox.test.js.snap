// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EuiCheckbox is rendered 1`] = `
<div
  class="euiCheckbox testClass1 testClass2"
>
  <input
    aria-label="aria-label"
    class="euiCheckbox__input"
    data-test-subj="test subject string"
    id="id"
    type="checkbox"
  />
  <div
    class="euiCheckbox__square"
  />
</div>
`;

exports[`EuiCheckbox props check is rendered 1`] = `
<div
  class="euiCheckbox"
>
  <input
    checked=""
    class="euiCheckbox__input"
    id="id"
    type="checkbox"
  />
  <div
    class="euiCheckbox__square"
  />
</div>
`;

exports[`EuiCheckbox props disabled disabled is rendered 1`] = `
<div
  class="euiCheckbox"
>
  <input
    class="euiCheckbox__input"
    disabled=""
    id="id"
    type="checkbox"
  />
  <div
    class="euiCheckbox__square"
  />
</div>
`;

exports[`EuiCheckbox props label is rendered 1`] = `
<div
  class="euiCheckbox"
>
  <input
    class="euiCheckbox__input"
    id="id"
    type="checkbox"
  />
  <div
    class="euiCheckbox__square"
  />
  <label
    class="euiCheckbox__label"
    for="id"
  >
    <span>
      Label
    </span>
  </label>
</div>
`;

exports[`EuiCheckbox props type inList is rendered 1`] = `
<div
  class="euiCheckbox euiCheckbox--inList"
>
  <input
    class="euiCheckbox__input"
    id="id"
    type="checkbox"
  />
  <div
    class="euiCheckbox__square"
  />
</div>
`;
